File: solutions_5658571765186560_1/C++/Rongqi/prob_d.cpp
#include <iostream>
#include <vector>
using namespace std;

int testcases;

int main() {
	cin >> testcases;
	for (int testcase = 0; testcase < testcases; testcase ++) {
		int n, a, b;
		cin >> n >> a >> b;
		if (a < b) {
			int tmp = a; a = b; b = tmp;
		}
		bool o = false; 

		if (n <= 6) {
			if (n == 6) {
				o = (a >= 6) && (b >= 3) && ((a * b) % 6 == 0);
			} else if (n == 5) {
				o = (a >= 5) && (b >= 4) && ((a * b) % 5 == 0);
			} else if (n == 4) {
				o = (a >= 4) && (b >= 3) && ((a * b) % 4 == 0);
			} else if (n == 3) {
				o = (a >= 3) && (b >= 2) && ((a * b) % 3 == 0);
			} else if (n == 2) {
				o = (a >= 2) && (b >= 1) && ((a * b) % 2 == 0);
			} else if (n == 1) {
				o = true;
			}
		}

		cout << "Case #" << testcase + 1 << ": ";
		if (o) {
			cout << "GABRIEL";
		} else {
			cout << "RICHARD";
		}

		cout << endl;
	}

	return 0;
}

File: solutions_5658571765186560_0/Python/handorff/probd.py
gabe = 'GABRIEL'
rick = 'RICHARD'

def solve(x, r, c):
    if r < c:
        tmp = r
        r = c
        c = tmp
    if x == 1:
        return gabe
    elif x == 2:
        if (r * c) % 2 == 0:
            return gabe
        else:
            return rick
    elif x == 3:
        if r != 3 and c != 3:
            return rick
        elif (r, c) == (3, 1):
            return rick
        else:
            return gabe
    elif x == 4:
        if (r * c) % 4 != 0:
            return rick
        elif (r, c) in [(4, 3), (4, 4)]:
            return gabe
        else:
            return rick
    else:
        assert 0, 'X not in [1, 4]'

t = int(raw_input())
test_cases = []

for i in xrange(t):
    x, r, c = map(int, raw_input().split(' '))
    test_cases.append((x, r, c))

for i in xrange(t):
    x, r, c = test_cases[i]
    print "Case #%d: %s" % (i + 1, solve(x, r, c))
